impl/src/main/java/org/ehcache/internal/store/heap/OnHeapStore.java

```diff
@@ -271,30 +298,40 @@ public void put(final K key, final V value) throws CacheAccessException {
 
     final AtomicBoolean entryActuallyAdded = new AtomicBoolean();
     final long now = timeSource.getTimeMillis();
-
+    
+    final AtomicLong replacedValueSize = new AtomicLong(0l);
     try {
       OnHeapValueHolder<V> valuePut = map.compute(key, new BiFunction<K, OnHeapValueHolder<V>, OnHeapValueHolder<V>>() {
         @Override
         public OnHeapValueHolder<V> apply(K mappedKey, OnHeapValueHolder<V> mappedValue) {
-          entryActuallyAdded.set(mappedValue == null);
 
           if (mappedValue != null && mappedValue.isExpired(now, TimeUnit.MILLISECONDS)) {
             mappedValue = null;
           }
 
+          OnHeapValueHolder<V> toReturn = null;
           if (mappedValue == null) {
-            return newCreateValueHolder(key, value, now);
+            toReturn = newCreateValueHolder(key, value, now);
           } else {
-            return newUpdateValueHolder(key, mappedValue, value, now);
+            replacedValueSize.set(sizeOfEngine.sizeof(mappedValue.value()));// get it only for mapped value
+            toReturn = newUpdateValueHolder(key, mappedValue, value, now);
           }
+          entryActuallyAdded.set(toReturn != null);
```

Comment:
This completely breaks the put statistics as _replaced_ will no longer be triggered. And thus I believe it breaks size accounting as well since it depends on the same test.